include/grpcpp/security/credentials.h

```diff
@@ -62,8 +75,23 @@ class ChannelCredentials : private GrpcLibraryCodegen {
       const std::shared_ptr<ChannelCredentials>& creds,
       const ChannelArguments& args);
 
+  friend std::shared_ptr<Channel>
+  experimental::CreateCustomChannelWithInterceptors(
+      const grpc::string& target,
+      const std::shared_ptr<ChannelCredentials>& creds,
+      const ChannelArguments& args,
+      std::unique_ptr<std::vector<
+          std::unique_ptr<experimental::ClientInterceptorFactoryInterface>>>
+          interceptor_creators);
+
   virtual std::shared_ptr<Channel> CreateChannel(
       const grpc::string& target, const ChannelArguments& args) = 0;
+
+  virtual std::shared_ptr<Channel> CreateChannelWithInterceptors(
+      const grpc::string& target, const ChannelArguments& args,
+      std::unique_ptr<std::vector<
+          std::unique_ptr<experimental::ClientInterceptorFactoryInterface>>>
+          interceptor_creators) = 0;
```

Comment:
You can't add a pure virtual function to a public (non-internal) interface even if it is in the private section (API breaker). You can just make it return nullptr in the default implementation, but add a comment saying that this should never happen and that this is just in place because this is post-1.0 API.

Reply:
done